network/discovery/discovery.go

```diff
@@ -0,0 +1,426 @@
+package discovery
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"github.com/0xPolygon/polygon-edge/network/common"
+	"github.com/0xPolygon/polygon-edge/network/event"
+	"github.com/hashicorp/go-hclog"
+	"github.com/libp2p/go-libp2p-core/network"
+	"time"
+
+	"github.com/0xPolygon/polygon-edge/network/grpc"
+	"github.com/0xPolygon/polygon-edge/network/proto"
+	"github.com/libp2p/go-libp2p-core/peer"
+	kb "github.com/libp2p/go-libp2p-kbucket"
+)
+
+const (
+	// maxDiscoveryPeerReqCount is the max peer count that
+	// can be requested from other peers
+	maxDiscoveryPeerReqCount = 16
+
+	// peerDiscoveryInterval is the interval at which other
+	// peers are queried for their peer sets
+	peerDiscoveryInterval = 5 * time.Second
+
+	// bootnodeDiscoveryInterval is the interval at which
+	// random bootnodes are dialed for their peer sets
+	bootnodeDiscoveryInterval = 60 * time.Second
+)
+
+// networkingServer defines the base communication interface between
+// any networking server implementation and the DiscoveryService
+type networkingServer interface {
+	// BOOTNODE QUERIES //
+
+	// IsBootnode checks if the peer is a registered bootnode
+	IsBootnode(peerID peer.ID) bool
+
+	// GetRandomBootnode fetches a random bootnode, if any
+	GetRandomBootnode() *peer.AddrInfo
+
+	// GetBootnodeConnCount fetches the number of bootnode connections [Thread safe]
+	GetBootnodeConnCount() int64
+
+	// PROTOCOL MANIPULATION //
+
+	// NewDiscoveryClient returns a discovery gRPC client connection
+	NewDiscoveryClient(peerID peer.ID) (proto.DiscoveryClient, error)
+
+	// CloseProtocolStream closes a protocol stream to the peer
+	CloseProtocolStream(protocol string, peerID peer.ID) error
+
+	// PEER MANIPULATION //
+
+	// DisconnectFromPeer attempts to disconnect from the specified peer
+	DisconnectFromPeer(peerID peer.ID, reason string)
+
+	// AddToPeerStore adds a peer to the networking server's peer store
+	AddToPeerStore(peerInfo *peer.AddrInfo)
+
+	// GetPeerInfo fetches the peer information from the server's peer store
+	GetPeerInfo(peerID peer.ID) *peer.AddrInfo
+
+	// GetRandomPeer fetches a random peer from the server's peer store
+	GetRandomPeer() *peer.ID
+
+	// TEMPORARY DIALING //
+
+	// FetchAndSetTemporaryDial checks if the peer connection is a temporary dial,
+	// and sets a new value accordingly
+	FetchAndSetTemporaryDial(peerID peer.ID, newValue bool) bool
```

Comment:
I find this naming to be a little misleading since `newValue` will be set once initially and loaded on subsequent calls. Maybe change it to `FetchOrSetTemporaryDial` ?

Reply:
Completely agree 👍 

Fixed the naming here

[d2f3bc8](https://github.com/0xPolygon/polygon-edge/pull/343/commits/d2f3bc89f9bd4607c77813a0ddd679607f31d939)